lib/resources/groups.rb

```diff
@@ -14,7 +14,11 @@ module GroupManagementSelector
     # returns nil, if no group manager was found for the operating system
     def select_group_manager(os)
       if os.unix?
-        @group_provider = UnixGroup.new(inspec)
+        @group_provider = if os.darwin?
```

Comment:
Is there a reason to nest this under a `os.unix?` check? I think it might be cleaner to do this:

```ruby
if os.darwin?
  @group_provider = DarwinGroup.new(inspec)
elsif os.unix?
  @group_provider = UnixGroup.new(inspec)
elsif os.windows?
  @group_provider = WindowsGroup.new(inspec)
end
```

... or, if you want to use the conditional assignment stuff:

```ruby
@group_provider = if os.darwin?
                    DarwinGroup.new(inspec)
                  elsif os.unix?
                    UnixGroup.new(inspec)
                  elsif os.windows?
                    WindowsGroup.new(inspec)
                  end
```

Reply:
I can see that, I originally did it that way because `os.unix?` was at the top and triggering first. I suppose we could be `os.darwin?` first and achieve the same outcome.

I think the latter is cleaner so I'll do that.